aten/src/ATen/native/Normalization.cpp

```diff
@@ -199,6 +199,7 @@ std::tuple<Tensor,Tensor> batch_norm_cpu_update_stats_template(
   using accscalar_t = at::acc_type<scalar_t, false>;
 
   int64_t n_input = input.size(1);
+  TORCH_CHECK(n_input > 0 && input.numel() > 0, "both input.size(1) and input.numel() need to be greater than 0, got input.size(1) = ", n_input, ", input.numel() = ", input.numel());
```

Comment:
Is there any case that: input.numel() >0 but n_input ==0 ?

Reply:
Yes, if `input = torch.rand(1, 0)`.

Reply:
If `input = torch.rand(1, 0)` the results are:
n_input : 0
input.numel() : 0

Reply:
My bad, typo, should be 
```
input = torch.rand(0, 1)


print("input.size(0): ", input.size(0))
print("input.size(1): ", input.size(1))
print("input.numel(): ", input.numel())
```
outputs
```
input.size(0):  0
input.size(1):  1
input.numel():  0
```

Reply:
@CaoE misread the question, sry. You are right, i dont think it is possible ` input.numel() >0 but n_input ==0`. input.numel() >0 implies all dimension has size>0.

Reply:
Do you suggest checking only `input.numel() >0`? Both input.numel() and n_input are used as denominator so it is safer to check both variables. Understand it might be redundent. 
Not sure if there are corner cases  with ` input.numel() >0 but n_input ==0` (i couldnt think one).

Reply:
There is also `int64_t image_size = input.numel() / input.size(0) / input.size(1);` in https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/cpu/batch_norm_kernel.cpp#L1354.
I think check `input.numel() >0` is enough or you need also check `input.size(0)`.